manifests/0000_70_cluster-network-operator_01_egr_crd.yaml

```diff
@@ -0,0 +1,128 @@
+spec:
+  group: network.operator.openshift.io
+  names:
+    kind: EgressRouter
+    listKind: EgressRouterList
+    plural: egressrouters
+    shortNames:
+    - egr
+    singular: egressrouter
+  scope: Namespaced
+  versions:
+  - additionalPrinterColumns:
+    - jsonPath: .status.egressRouter
+      name: EgressRouters
+      type: string
+    - jsonPath: .status.status
+      name: Status
+      type: string
+    name: v1
+    schema:
+      openAPIV3Schema:
+        description: EgressRouter is a single egressrouter pod configuration object.
+        properties:
+          apiVersion:
+            description: 'APIVersion defines the versioned schema of this representation
+              of an object. Servers should convert recognized schemas to the latest
+              internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources'
+            type: string
+          kind:
+            description: 'Kind is a string value representing the REST resource this
+              object represents. Servers may infer this from the endpoint the client
+              submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds'
+            type: string
+          metadata:
+            type: object
+          spec:
+            description: EgressRouterSpec contains the configuration for an egress
+              router.
+            properties:
+              ip:
+                description: IP configuration arguments.
+                properties:
+                  addresses:
+                    description: List of IP addresses to configure on the interface.
+                    items:
+                      description: EgressRouterAddress contains a pair of ip cidr
+                        and gateway to be configured on the router's interface
+                      properties:
+                        gateway:
+                          description: IP address of the next-hop gateway, if it cannot
+                            be automatically determined. Can be IPv4 or IPv6.
+                          type: string
+                        ip:
+                          description: IP is the address to configure on the router's
+                            interface. Can be IPv4 or IPv6.
+                          type: string
+                      required:
+                      - ip
+                      type: object
+                    type: array
+                  allowedDestinations:
+                    description: List of CIDR blocks that the pod is allowed to connect
+                      to via this interface. If not provided, the pod can connect
+                      to any destination.
+                    items:
+                      description: CIDR represents an IPv4 or IPV6 CIDR block.
+                      type: string
```

Comment:
Again a pattern would be nice, you can use this one for IPv4.
https://github.com/openshift/cluster-network-operator/blob/master/bindata/network/openshift-sdn/001-crd.yaml#L48

Pretty sure there is some regex somewhere for the IPv6 one.

Reply:
ditto

Reply:
I have recently been blocked by apiserver team for not using regex on IP fields:

https://github.com/openshift/api/pull/846

If this is a cidr, I believe you can just change type: string for type: cidr, and that makes the validation (cidr does the regex validation).

Reply:
Note that this has been quite changed, now its on https://github.com/openshift/api/pull/851 and stts has also mentioned that, so this whole PR is deending on that one, so don't consider this issue. Thanks!